exporter/awskinesisexporter/internal/key/key.go

```diff
@@ -5,12 +5,41 @@ package key // import "github.com/open-telemetry/opentelemetry-collector-contrib
 
 import (
 	"github.com/google/uuid"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/traceutil"
 )
 
-// Partition allows for switching our partitioning behavior
+// Partitioner allows for switching our partitioning behavior
 // when sending data to kinesis.
-type Partition func(v any) string
+type Partitioner interface {
+	Partition(v any) string
+	Keyed() bool
+}
+
+type Randomized struct{}
+
+func (p Randomized) Partition(_ any) string {
+	return uuid.NewString()
+}
+
+func (p Randomized) Keyed() bool {
+	return false
+}
+
+type TraceID struct{}
+
+func (p TraceID) Partition(v any) string {
+	if trace, ok := v.(ptrace.Traces); ok {
+		traceKey := traceutil.TraceIDToHexOrEmptyString(trace.ResourceSpans().At(0).ScopeSpans().At(0).Spans().At(0).TraceID())
```

Comment:
I would prefer to some length checks or ranges over the data since this has a high likelihood to get wrong and cause a panic.

Reply:
Once this is safe for any value, I would be happy to approve it :)